src/main/java/com/redhat/lightblue/hook/publish/IdentityExctractionUtil.java

```diff
@@ -0,0 +1,191 @@
+package com.redhat.lightblue.hook.publish;
+
+import static com.redhat.lightblue.hook.publish.model.IdentitySet.INSERT_OPERATION;
+import static com.redhat.lightblue.hook.publish.model.IdentitySet.UPDATE_OPERATION;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.skyscreamer.jsonassert.JSONParser;
+
+import com.redhat.lightblue.hook.publish.model.Identity;
+import com.redhat.lightblue.hook.publish.model.IdentitySet;
+import com.redhat.lightblue.hook.publish.model.JSONWrapper;
+
+public class IdentityExctractionUtil {
+
+    /*
+     * Accepts JSONObject / JSONArray and returns the IdentitySet for the
+     * required permutations.
+     * 
+     * each IdentitySet implies one event
```

Comment:
@paterczm there is a 1:1 relationship between those two, I think you are trying to imply that we could rename this to an event extraction util and return sets of events. if yes, I like that suggestion, lemme make the changes.